### DISCUSSION TITLE
Trying to read response outside of async with aiohttp.ClientSession() as session

### DISCUSSION THREAD

**Discussion by aruseni** [aio-libs/aiohttp#5813] [Q&A] [answered]
This issue was previously mentioned, see #4015.

If you do this, it just works fine.

```python
import aiohttp
import asyncio

async def main():

    async with aiohttp.ClientSession() as session:
        response = await session.get("https://docs.aiohttp.org/en/stable/")

        print("Status:", response.status)
        print("Content-type:", response.headers['content-type'])

        html = await response.text()
        print("Body:", html[:16], "...")

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

But if you try to read the response outside of the async with block, it will often fail.

```python
import aiohttp
import asyncio

async def main():

    async with aiohttp.ClientSession() as session:
        response = await session.get("https://docs.aiohttp.org/en/stable/")

    print("Status:", response.status)
    print("Content-type:", response.headers['content-type'])

    html = await response.text()
    print("Body:", html[:16], "...")

loop = asyncio.get_event_loop()
loop.run_until_complete(main())
```

What is interesting is that it does not always fail. If you try to run this code several times, then it is likely that at least one time it will also read the response. Does anyone know what causes it?

**Comment by webknjaz** [comment, marked as answer]
Async context manager makes sure to release the resources. You shouldn't access them outside of it.

**Reply by aruseni** [reply]
Thanks! I get it. But why does it sometimes work? Why does not this code fail every time?

**Reply by webknjaz** [reply]
I think that the async exit doesn't always close the connection instantly.
